src/common/obj_bencher.cc

```diff
@@ -443,8 +444,9 @@ int ObjBencher::write_bench(int secondsToRun,
   lock.Lock();
   data.finished = 0;
   data.start_time = ceph_clock_now();
+  stopTime = data.start_time + runtime;
   lock.Unlock();
-  for (int i = 0; i<concurrentios; ++i) {
+  for (int i = 0; i<concurrentios && ceph_clock_now() < stopTime; ++i) {
```

Comment:
@tchaikov 
Hi, 
- `aio_write()` is a non-blocking call, but if the `concurrentios` is much larger than the variable `<seconds>`, it will not sent completely in the time.  

- I've encounter this situation when one of the requests was blocked, the process would never be stopped.
 
```
while (data.finished < data.started) {
    slot = data.finished % concurrentios;
    completion_wait(slot);
    lock.Lock();
    r = completion_ret(slot);
    if (r != 0) {
      lock.Unlock();
      goto ERR;
    }
    data.cur_latency = ceph_clock_now(cct) - start_times[slot];
    data.history.latency.push_back(data.cur_latency);
    total_latency += data.cur_latency;
    if (data.cur_latency > data.max_latency) data.max_latency = data.cur_latency;
    if (data.cur_latency < data.min_latency) data.min_latency = data.cur_latency;
    ++data.finished;
    data.avg_latency = total_latency / data.finished;
    --data.in_flight;
    lock.Unlock();
    release_completion(slot);
    delete contents[slot];
    contents[slot] = 0;
  }
```
So, I think the upper module may need to be removed and delete the `contents[slot]` which is not completed. what's your opinion?